tests/util/test_network_protocol_files.py

```diff
@@ -0,0 +1,521 @@
+
+    message_bytes, input_bytes = parse_blob(input_bytes)
+    message = pool_protocol.PutFarmerResponse.from_bytes(message_bytes)
+    assert message == put_farmer_response
+    assert bytes(message) == bytes(put_farmer_response)
+
+    message_bytes, input_bytes = parse_blob(input_bytes)
+    message = pool_protocol.ErrorResponse.from_bytes(message_bytes)
+    assert message == error_response
+    assert bytes(message) == bytes(error_response)
+
+    return input_bytes
+
+
+def parse_timelord_protocol(input_bytes):
+    message_bytes, input_bytes = parse_blob(input_bytes)
+    message = timelord_protocol.NewPeakTimelord.from_bytes(message_bytes)
+    assert message == new_peak_timelord
+    assert bytes(message) == bytes(new_peak_timelord)
+
+    message_bytes, input_bytes = parse_blob(input_bytes)
+    message = timelord_protocol.NewUnfinishedBlockTimelord.from_bytes(message_bytes)
+    assert message == new_unfinished_block_timelord
+    assert bytes(message) == bytes(new_unfinished_block_timelord)
+
+    message_bytes, input_bytes = parse_blob(input_bytes)
+    message = timelord_protocol.NewInfusionPointVDF.from_bytes(message_bytes)
+    assert message == new_infusion_point_vdf
+    assert bytes(message) == bytes(new_infusion_point_vdf)
+
+    message_bytes, input_bytes = parse_blob(input_bytes)
+    message = timelord_protocol.NewSignagePointVDF.from_bytes(message_bytes)
+    assert message == new_signage_point_vdf
+    assert bytes(message) == bytes(new_signage_point_vdf)
+
+    message_bytes, input_bytes = parse_blob(input_bytes)
+    message = timelord_protocol.NewEndOfSubSlotVDF.from_bytes(message_bytes)
+    assert message == new_end_of_sub_slot_bundle
+    assert bytes(message) == bytes(new_end_of_sub_slot_bundle)
+
+    message_bytes, input_bytes = parse_blob(input_bytes)
+    message = timelord_protocol.RequestCompactProofOfTime.from_bytes(message_bytes)
+    assert message == request_compact_proof_of_time
+    assert bytes(message) == bytes(request_compact_proof_of_time)
+
+    message_bytes, input_bytes = parse_blob(input_bytes)
+    message = timelord_protocol.RespondCompactProofOfTime.from_bytes(message_bytes)
+    assert message == respond_compact_proof_of_time
+    assert bytes(message) == bytes(respond_compact_proof_of_time)
+
+    return input_bytes
+
+
+class TestNetworkProtocolFiles:
+    def test_network_protocol_files(self):
+        filename = get_network_protocol_filename()
+        assert os.path.exists(filename)
+        with open(filename, "rb") as f:
+            input_bytes = f.read()
+        input_bytes = parse_farmer_protocol(input_bytes)
+        input_bytes = parse_full_node_protocol(input_bytes)
+        input_bytes = parse_wallet_protocol(input_bytes)
+        input_bytes = parse_harvester_protocol(input_bytes)
+        input_bytes = parse_introducer_protocol(input_bytes)
+        input_bytes = parse_pool_protocol(input_bytes)
+        input_bytes = parse_timelord_protocol(input_bytes)
```

Comment:
it would seem more appropriate (to me) to have separate test cases for each message, and rather than reading a file from disk and parsing the messages in the same order they were written (which is a possible source of errors), the tests cases could just call the functions that produce the messages directly.

I think that would simplify this test.

The benefit of having separate test cases for each message is that if one message fails, there's no avalanche causing other messages to fail as well. They are independent.